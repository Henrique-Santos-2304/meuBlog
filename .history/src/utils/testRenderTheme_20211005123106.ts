import { ThemeProvider } from "styled-components";
import { render, RenderResult } from "@testing-library/react";
import theme from "../styles/theme";
export const renderTheme = (children: React.ReactNode): RenderResult => {
  return render(<ThemeProvider>{children}</ThemeProvider> );
};
