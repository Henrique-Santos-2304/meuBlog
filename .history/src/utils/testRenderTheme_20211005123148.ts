import { ThemeProvider } from "styled-components";
import { render, RenderResult } from "@testing-library/react";
import theme from "../styles/theme";
export const renderTheme = (children: React.ReactElement): RenderResult => {
  return render(<ThemeProvider theme={theme}>{children}</ThemeProvider> );
};
