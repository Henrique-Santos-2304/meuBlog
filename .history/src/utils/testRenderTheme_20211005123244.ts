import { ThemeProvider } from "styled-components";
import { render, RenderResult } from "@testing-library/react";
import theme from "../styles/theme";
export const renderTheme = (children) => {
  return render(<ThemeProvider theme={theme}>{children}</ThemeProvider> );
};
