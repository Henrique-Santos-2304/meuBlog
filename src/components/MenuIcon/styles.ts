import styled, { css, DefaultTheme } from "styled-components";

import { CheckProps } from ".";

const modifierStyleMenu = {
  checkIn: (theme: DefaultTheme) => css`
    background: ${theme.colors.red800};
    transform: rotate(45deg);
    &::after,
    &::before {
      transform: rotate(90deg);
      background: ${theme.colors.red800};
    }

    &::after {
      bottom: 0;
    }
    &::before {
      top: 0;
    }
  `,
  checkOut: (theme: DefaultTheme) => css`
    background: ${theme.colors.white};
    &::after {
      background: ${theme.colors.white};
      bottom: -0.6rem;
    }
    &::before {
      background: ${theme.colors.white};
      top: -0.6rem;
    }
  `,
};

export const Wrapper = styled.div<CheckProps>`
  ${({ theme }) => css`

    display: flex;
    justify-content: center;
    align-items: center;
    cursor: pointer;

    width: 4.125rem;
    height: 4.125rem;
    border-radius: 50%;

    background: ${theme.colors.mainBg};
    border: 0.155rem solid transparent;
    transition: 0.2s linear;
    &:hover {
      width: 4.65rem;
      height: 4.65rem;
  `}
`;

export const LineSpan = styled.span<CheckProps>`
  ${({ check, theme }) => css`
    display: block;
    width: 1.8rem;
    margin: 0 1rem;
    height: 0.2rem;
    position: relative;
    transition: 0.4s ease-in-out;
    ${check
      ? modifierStyleMenu.checkIn(theme)
      : modifierStyleMenu.checkOut(theme)}

    &::after,
    &::before {
      content: "";
      display: block;
      width: 100%;
      height: 100%;
      position: absolute;
      transition: 0.4s ease-in-out;
    }
  `}
`;
