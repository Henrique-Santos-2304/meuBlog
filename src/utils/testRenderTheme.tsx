import { ThemeProvider } from "styled-components";
import { ChakraProvider } from "@chakra-ui/react";

import { render, RenderResult } from "@testing-library/react";
import theme from "styles/theme";
export const renderTheme = (children: React.ReactNode): RenderResult => {
  return render(
    <ThemeProvider theme={theme}>
      <ChakraProvider> {children}</ChakraProvider>
    </ThemeProvider>
  );
};
