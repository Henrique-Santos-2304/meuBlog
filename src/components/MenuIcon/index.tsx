import * as S from "./styles";

export interface CheckProps {
  check?: boolean;
  handleOpenMenu?: () => void;
}
const MenuIcon = ({ check, handleOpenMenu }: CheckProps) => {
  return (
    <S.Wrapper data-testid="divClick" onClick={handleOpenMenu} check={check}>
      <S.LineSpan data-testid="span-test" check={check}></S.LineSpan>
    </S.Wrapper>
  );
};

export default MenuIcon;
