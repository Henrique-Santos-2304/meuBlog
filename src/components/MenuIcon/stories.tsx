import { Story, Meta } from "@storybook/react";
import MenuIcon from ".";

export default {
  title: "MenuIcon",
  component: MenuIcon,
} as Meta;

export const Basic: Story = (args) => <MenuIcon {...args} check={false} />;

Basic.args = {
  title: "title basic",
  description: "description basic",
  color: "#111",
  background: "#111",
};

export const Default: Story = (args) => <MenuIcon {...args} check={false} />;
