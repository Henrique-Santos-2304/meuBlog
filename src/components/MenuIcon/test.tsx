// import { screen } from "@testing-library/react";
import React from "react";
import { renderTheme } from "utils/testRenderTheme";
import { screen } from "@testing-library/react";
import "jest-styled-components";

import MenuIcon from ".";

describe("<Teste />", () => {
  it("should render ", () => {
    const { container } = renderTheme(<MenuIcon check={false} />);

    expect(container).toMatchSnapshot();
  });
  it("should change style with event hover element ", () => {
    renderTheme(<MenuIcon check={false} />);
    const buttonIcon = screen.getByRole("button", { hidden: true });
    expect(buttonIcon).toHaveStyle({
      width: "4.125rem",
      height: "4.125rem",
      transition: "0.2s linear",
    });

    expect(buttonIcon).toHaveStyleRule("width", "4.65rem", {
      modifier: ":hover",
    });
    expect(buttonIcon).toHaveStyleRule("height", "4.65rem", {
      modifier: ":hover",
    });
  });
  it("hope i have them when menu is closed", () => {
    renderTheme(<MenuIcon check={false} />);
    const spanMenuIcon = screen.getByTestId("span-test");
    expect(spanMenuIcon).toHaveStyle({
      background: "#E5E7EB",
    });

    expect(spanMenuIcon).toHaveStyleRule("background", "#E5E7EB", {
      modifier: "::before",
    });
    expect(spanMenuIcon).toHaveStyleRule("top", "-0.6rem", {
      modifier: "::before",
    });
    expect(spanMenuIcon).toHaveStyleRule("bottom", "-0.6rem", {
      modifier: "::after",
    });
    expect(spanMenuIcon).toHaveStyleRule("background", "#E5E7EB", {
      modifier: "::after",
    });
  });

  it("Hope you have these styles when the input is  checked", () => {
    renderTheme(<MenuIcon check={true} />);
    const spanMenuIcon = screen.getByTestId("span-test");
    expect(spanMenuIcon).toHaveStyle({
      background: "#C11010",
    });
    expect(spanMenuIcon).toHaveStyleRule("background", "#C11010", {
      modifier: "::before",
    });
    expect(spanMenuIcon).toHaveStyleRule("background", "#C11010", {
      modifier: "::after",
    });
    expect(spanMenuIcon).toHaveStyleRule("top", "0", {
      modifier: "::before",
    });
    expect(spanMenuIcon).toHaveStyleRule("bottom", "0", {
      modifier: "::after",
    });
  });
});
