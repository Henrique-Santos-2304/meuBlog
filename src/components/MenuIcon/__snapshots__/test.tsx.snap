// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Teste /> should render  1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: pointer;
  width: 4.125rem;
  height: 4.125rem;
  border-radius: 50%;
  background: #12131C;
  border: 0.155rem solid transparent;
  -webkit-transition: 0.2s linear;
  transition: 0.2s linear;
}

.c0:hover {
  width: 4.65rem;
  height: 4.65rem;
}

.c1 {
  display: block;
  width: 1.8rem;
  margin: 0 1rem;
  height: 0.2rem;
  position: relative;
  -webkit-transition: 0.4s ease-in-out;
  transition: 0.4s ease-in-out;
  background: #E5E7EB;
}

.c1::after {
  background: #E5E7EB;
  bottom: -0.6rem;
}

.c1::before {
  background: #E5E7EB;
  top: -0.6rem;
}

.c1::after,
.c1::before {
  content: "";
  display: block;
  width: 100%;
  height: 100%;
  position: absolute;
  -webkit-transition: 0.4s ease-in-out;
  transition: 0.4s ease-in-out;
}

<div>
  <div
    class="c0"
    data-testid="divClick"
  >
    <span
      class="c1"
      data-testid="span-test"
    />
  </div>
</div>
`;
